contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -350,12 +351,11 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         // Store the receipt of this message delivery. When a subsquent message is sent back
         // to the origin of this message, we will clean up the receipt state.
         // If the receipts queue contract for this chain doesn't exist yet, create it now.
-        ReceiptQueue receiptsQueue = outstandingReceipts[
+        ReceiptQueue.TeleporterMessageReceiptQueue storage receiptsQueue = outstandingReceipts[
             warpMessage.originChainID
         ];
-        if (address(receiptsQueue) == address(0)) {
-            receiptsQueue = new ReceiptQueue();
-            outstandingReceipts[warpMessage.originChainID] = receiptsQueue;
+        if (receiptsQueue.owner == address(0)) {
+            receiptsQueue.owner = msg.sender;
```

Comment:
in this case is the first relayer that delivers a Teleporter message to this chain set as the "owner" of the receipt queue?

Reply:
removed the owner